test/Infrastructure/LeanCode.AuditLogs.Tests/AuditLogsFilterTests.cs

```diff
@@ -0,0 +1,175 @@
+using System.Diagnostics;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using FluentAssertions;
+using LeanCode.CQRS.MassTransitRelay;
+using LeanCode.OpenTelemetry;
+using LeanCode.TimeProvider;
+using LeanCode.TimeProvider.TestHelpers;
+using MassTransit;
+using MassTransit.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using Xunit;
+
+namespace LeanCode.AuditLogs.Tests;
+
+public sealed class AuditLogsFilterTests
+{
+    private const string SomeId = "some_id";
+    private const string ActorId = "actor_id";
+
+    private static readonly JsonSerializerOptions Options =
+        new()
+        {
+            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+            ReferenceHandler = ReferenceHandler.IgnoreCycles,
+            WriteIndented = false,
+        };
+
+    private readonly IServiceProvider serviceProvider;
+    private readonly ITestHarness harness;
+    private static readonly TestEntity TestEntity = new() { Id = SomeId };
+
+    public AuditLogsFilterTests()
+    {
+        TestTimeProvider.ActivateFake(new DateTimeOffset(2023, 10, 6, 11, 0, 3, 0, TimeSpan.Zero));
+        var collection = new ServiceCollection();
+        collection
+            .AddOpenTelemetry()
+            .ConfigureResource(r => r.AddService("AuditLogsFilterTests", serviceInstanceId: Environment.MachineName))
+            .WithTracing(builder =>
+            {
+                builder
+                    .AddProcessor<IdentityTraceAttributesFromBaggageProcessor>()
+                    .AddSource("MassTransit")
+                    .AddLeanCodeTelemetry();
+            });
+        collection.AddDbContext<TestDbContext>();
+        collection.AddTransient<IAuditLogStorage, StubAuditLogStorage>();
+        collection.AddMassTransitTestHarness(ConfigureMassTransit);
+
+        serviceProvider = collection.BuildServiceProvider();
+        harness = serviceProvider.GetRequiredService<ITestHarness>();
+    }
+
+    private static void ConfigureMassTransit(IBusRegistrationConfigurator cfg)
+    {
+        cfg.AddConsumersWithDefaultConfiguration(
+            new[] { typeof(Consumer).Assembly, typeof(AuditLogsConsumer).Assembly },
+            typeof(DefaultConsumerDefinition<>)
+        );
+
+        cfg.UsingInMemory(
+            (ctx, busCfg) =>
+            {
+                busCfg.ConfigureEndpoints(ctx, new DefaultEndpointNameFormatter("InMemory"));
+                busCfg.ConnectBusObservers(ctx);
+            }
+        );
+    }
+
+    [Fact]
+    public async Task Publishes_entity_changes_to_bus()
+    {
+        await harness.Start();
```

Comment:
Let's move this call to the `StartAsync` of `IAsyncLifetime`.

Reply:
And probably the `consumerHarness` can be shared across tests (i.e. set it in ctor as a class field).